tests/authenticatedAddingAndRemoving.spec.cjs

```diff
@@ -0,0 +1,176 @@
+	await page.locator(".btn.btn-primary").click();
+	await expect(page.getByText("login successfully")).toBeVisible();
+
+    //add 2 items to cart
+    const productTitleLocator = page.getByText(productName);
+	const parentLocator = productTitleLocator.locator('..').locator('..');
+	const firstButton = parentLocator.locator('div > button').filter({hasText: 'ADD TO CART'});
+	await expect(firstButton).toBeVisible();
+	await firstButton.click();
+
+	const productTitleLocator2 = page.getByText(productName2);
+	const parentLocator2 = productTitleLocator2.locator('..').locator('..');
+	const secondButton = parentLocator2.locator('div > button').filter({hasText: 'ADD TO CART'});
+	await expect(secondButton).toBeVisible();
+	await secondButton.click();
+
+	await expect(page.getByText("Item Added To cart").first()).toBeVisible();
+
+	//go to cart page
+	await page.getByRole("link", { name: "CART" }).click();
+
+    //ensure correct rendering of items in cart
+	await expect(page.getByText(productName, { exact: true })).toBeVisible();
+	await expect(page.getByText(productName2, { exact: true })).toBeVisible();
+	await expect(
+		page.getByText("this is the best book", { exact: true })
+	).toBeVisible();
+	await expect(
+		page.getByText("this is the best book2", { exact: true })
+	).toBeVisible();
+	await expect(page.getByText("price : 18")).toBeVisible();
+	await expect(page.getByText("price : 19")).toBeVisible();
+
+
+    const productTitleLocator3 = page.getByText(productName);
+	const parentLocator3 = productTitleLocator3.locator('..').locator('..');
+	const removeButton = parentLocator3.locator('div > button').filter({hasText: 'Remove'});
+	await expect(removeButton).toBeVisible();
+	await removeButton.click();
+
+    //ensure correct rendering of items in cart
+	await expect(page.getByText(productName2, { exact: true })).toBeVisible();
+	await expect(
+		page.getByText("this is the best book2", { exact: true })
+	).toBeVisible();
+	await expect(page.getByText("price : 19")).toBeVisible();
+
+    //log out
+    await page.getByRole("button", { name: "TESTING NAME" }).click();
+    await page.getByRole("link", { name: "LOGOUT" }).click();
+
+    //entering details to log in
+	await page.getByPlaceholder("Enter Your Email").fill(email);
+	await page.getByPlaceholder("Enter Your Password").fill("test@test.com");
+	await page.locator(".btn.btn-primary").click();
+	await expect(page.getByText("login successfully")).toBeVisible();
+
+    await page.waitForTimeout(1000);
+    await page.waitForLoadState();
+    //go to cart page
+	await page.getByRole("link", { name: "CART" }).click();
+	
+    //ensure correct rendering of items in cart
+	await expect(page.getByText(productName2, { exact: true })).toBeVisible();
+	await expect(
+		page.getByText("this is the best book2", { exact: true })
+	).toBeVisible();
+	await expect(page.getByText("price : 19")).toBeVisible();
+});
+
+function generateRandomProductName() {
```

Comment:
Might be better to separate helper functions into their own file? Just my thoughts :))

Reply:
I agree with you helper function should have their own file. But for this function it is being used once thus leaving it in the same file.